pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -140,24 +140,27 @@ func (t *ResolvedPipelineRunTask) checkParentsDone(facts *PipelineRunFacts) bool
 // (4) Pipeline is in stopping state (one of the PipelineTasks failed)
 // Note that this means Skip returns false if a conditionCheck is in progress
 func (t *ResolvedPipelineRunTask) Skip(facts *PipelineRunFacts) bool {
-	// finally tasks are never skipped. If this is a final task, return false
-	if facts.isFinalTask(t.PipelineTask.Name) {
+	if facts.isFinalTask(t.PipelineTask.Name) || t.IsStarted() {
 		return false
 	}
 
-	// it already has TaskRun associated with it - PipelineTask not skipped
-	if t.IsStarted() {
-		return false
+	if t.conditionsSkip() || t.whenExpressionsSkip(facts) || t.parentTasksSkip(facts) || facts.IsStopping() {
+		return true
 	}
 
-	// Check if conditionChecks have failed, if so task is skipped
```

Comment:
NIT: would appreciate these comments back 🙏

Reply:
found them to be repetitive because they're all mentioned at the top of the function here: 
https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L136-L142